src/reducers/linter.tsx

```diff
@@ -177,7 +177,12 @@ export const fetchLinterMessages = ({
   url: string;
   versionId: number;
 }): ThunkActionCreator => {
-  return async (dispatch) => {
+  return async (dispatch, getState) => {
+    // See: https://github.com/mozilla/addons-code-manager/issues/591
+    if (getState().linter.isLoading) {
```

Comment:
This should only abort loading if `getState().forVersionId === versionId` is also true. Consider the compare page. Imagine that the user is viewing a version but the API for linter messages is slow to respond. Before it responds, the user switches to a new version. This should trigger a re-fetch of the linter messages. This seems likely because one could switch versions in the select box quickly.

Reply:
Thanks, I missed this condition. I wonder why we did not have it in the `LinterProvider` code though, any ideas?

Reply:
> Thanks, I missed this condition. I wonder why we did not have it in the `LinterProvider` code though, any ideas?

Because the mapper already factored in version mismatches when creating `messageMap` (it would be undefined for mismatches): 

https://github.com/mozilla/addons-code-manager/blob/50be0327f1ae670d743a570855d23bef3eff14c3/src/components/LinterProvider/index.tsx#L105